Dockerfile

```diff
@@ -0,0 +1,46 @@
+ARG CUDA_VERSION="11.8.0"
+ARG CUDNN_VERSION="8"
+ARG UBUNTU_VERSION="22.04"
+ARG PYTHON_VERSION="3.11.4"
+ARG DOCKER_FROM=nvidia/cuda:$CUDA_VERSION-cudnn$CUDNN_VERSION-devel-ubuntu$UBUNTU_VERSION
+
+
+
+FROM $DOCKER_FROM AS base
+
+RUN apt-get update && apt-get install -y \
+    wget \
+    bzip2 \
+    git \
+    ca-certificates \
+    --no-install-recommends && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+WORKDIR /tmp
+
+RUN wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && \
+    bash Miniconda3-latest-Linux-x86_64.sh -b -p /opt/conda && \
+    rm Miniconda3-latest-Linux-x86_64.sh
+
+ENV PATH="/opt/conda/bin:${PATH}"
+
+ENV MAX_LOCAL_CTX="4096"
+
+RUN conda create -y --name interpreter-default python="$PYTHON_VERSION"
+RUN echo "source activate interpreter-default" > ~/.bashrc
+WORKDIR /app
```

Comment:
I suggest, though it may not working, swapping to a cuda-runtime image and then copying conda config over to get a reduced final image size, I use that in my own images for text-generation-webui:

FROM nvidia/cuda:11.8.0-runtime-ubuntu22.04

\# Copy conda and cuda files over
COPY --from=builder /opt/conda /opt/conda
COPY --from=builder /usr/local/cuda-11.8/targets/x86_64-linux/include /usr/local/cuda-11.8/targets/x86_64-linux/include 

ENV PATH=/opt/conda/bin:$PATH

Reply:
I also run the following for maximum compatibility, has worked well for me so far:

`RUN apt-get -y update && apt-get -y install wget && wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/cuda-keyring_1.0-1_all.deb \
    && dpkg -i cuda-keyring_1.0-1_all.deb && apt-get update && apt-get upgrade -y \
    && apt-get -y install python3 build-essential \
    && apt-get -y install cuda-11.8 && apt-get -y install cuda-11.8 \
    && systemctl enable nvidia-persistenced \
    && cp /lib/udev/rules.d/40-vm-hotadd.rules /etc/udev/rules.d \
    && sed -i '/SUBSYSTEM=="memory", ACTION=="add"/d' /etc/udev/rules.d/40-vm-hotadd.rules
`

Reply:
note that I also had to run 

`RUN apt-get update && apt-get remove --purge -y nvidia-* \
    && apt-get install -y --allow-downgrades nvidia-driver-535/jammy-updates`

because there were some annoying incompatibilities with the nvidia cuda image drivers and the end user drivers

Reply:
i keep it my way so far .. the container is just for maximum compatibility .. 
no need to get N new installs or add udev rules